src/hsl.rs

```diff
@@ -150,15 +167,21 @@ impl<Wp, T> FromColor<Wp, T> for Hsl<Wp, T>
             hue: h.into(),
             saturation: s,
             lightness: l,
-            white_point: PhantomData,
+            space: PhantomData,
         }
     }
 
-    fn from_hsl(hsl: Hsl<Wp, T>) -> Self {
-        hsl
+    fn from_hsl<Sp: RgbSpace<WhitePoint=Wp>>(hsl: Hsl<Sp, T>) -> Self {
+        if TypeId::of::<Sp::Primaries>() == TypeId::of::<S::Primaries>() {
```

Comment:
Would it be better to add RgbSpace bound to the FromColor trait? It will remove the need for reinterpret_as function.

```
pub trait FromColor<S, Wp, T> for Hsl<S, T>
     where T: Float,
        S: RgbSpace<WhitePoint=Wp>
        Wp: WhitePoint,
```